lib/proxywrap.js

```diff
@@ -138,7 +199,21 @@ exports.proxy = function(iface, options) {
 
 		function onReadable() {
 			var chunk;
-			while (null != (chunk = socket.read())) {
+			chunk = socket.read();
+
+			if(null === chunk && header.length === 0){
```

Comment:
Can you please add here a non-strict condition filter?
`! options.strict && null === chunk && header.length === 0`